### PR TITLE
chore: remove duplicated ids from augmented images

### PR DESCRIPTION
**Pull request by anthonycorletti**
* [x] I have added tests to `tests` to cover my changes.
* [ ] I have updated `docs/`, if necessary.
* [ ] I have updated the `README.md`, if necessary.

***What existing issue does this pull request close?***

Not referenced in GH.

***How are these changes tested?***

Via pytest.

***Demonstration***

See tests.

***Provide additional context.***

Coming soon.

### CODE REVIEW

File: dataquality/loggers/model_logger/image_classification.py [hunk 1]
```diff
@@ -33,3 +34,53 @@ def __init__(
...
+        """
+        if len(model_output["id"]):
+            return super().write_model_output(model_output)
+
+    def _get_data_dict(self) -> Dict[str, Any]:
+        # Handle the binary case by converting it to 2-class classification
+        self._filter_duplicate_ids()
+
+        return super()._get_data_dict()
+        return super()._get_data_dict()
```
Comment (franz101) [inline dataquality/loggers/model_logger/image_classification.py:86]: Is this twice now?

File: dataquality/loggers/logger_config/image_classification.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 from dataquality.loggers.logger_config.text_classification import (
     TextClassificationLoggerConfig,
 )
 
 
 class ImageClassificationLoggerConfig(TextClassificationLoggerConfig):
-    ...
+    # Keep track of the ids that have been observed in the current epoch
+    # the key is the split and epoch like observed_ids["train_0"] = {0, 1, 2, 3}
+    observed_ids: Dict[str, set] = dict()
```
Comment (franz101) [inline dataquality/loggers/logger_config/image_classification.py:11]: Looks good to me. Is there anything we good to do to clear the observed_ids?

File: dataquality/loggers/model_logger/image_classification.py [hunk 3]
```diff
@@ -33,3 +34,53 @@ def __init__(
...
+        In image classification, it is possible that after filtering
+        duplicate IDs, there are none to write. In that case,
+        we'll get an error trying to write them, so we skip
+        """
+        if len(model_output["id"]):
+            return super().write_model_output(model_output)
+
+    def _get_data_dict(self) -> Dict[str, Any]:
+        # Handle the binary case by converting it to 2-class classification
+        self._filter_duplicate_ids()
```
Comment (franz101) [inline dataquality/loggers/model_logger/image_classification.py:83]: why no more lock?

**Review by franz101 — COMMENTED** [review]
So the problem lied in the test?
Was it because the ids were already obsessed and not reseted?

File: dataquality/loggers/model_logger/image_classification.py (hunk 1 above)
Comment (anthonycorletti) [reply dataquality/loggers/model_logger/image_classification.py:86]: whoops! typo

**Comment by anthonycorletti** [comment]
> So the problem lied in the test? Was it because the ids were already obsessed and not reseted?

it seems to matter in what order the tests are run. i realized that the first test to run after `tests/integrations/torch/test_text_classification_pt.py` is the only test to consistently be the intermittent failing test (i know that sounds confusing). i dont think its related to resetting ids specifically.

File: dataquality/loggers/logger_config/image_classification.py (hunk 2 above)
Comment (anthonycorletti) [reply dataquality/loggers/logger_config/image_classification.py:11]: @franz101 https://github.com/rungalileo/dataquality/pull/485/commits/4b73aed2fa6efd6311713637c72b1688bd6c4939

**Review by franz101 — APPROVED** [review]

**Pull request merged**